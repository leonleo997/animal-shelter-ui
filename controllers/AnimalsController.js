import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            url: `animal`,
            baseURL: process.env.API,
            data: animal,
            method: 'POST'
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(animalName) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${animalName}`
        });
    }
}

