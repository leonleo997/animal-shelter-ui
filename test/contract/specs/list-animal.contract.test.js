import { provider } from "./init-pact";
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";
import { } from '@pact-foundation/pact'; 

describe('Given an Animal Service', () => {
    describe('When a request to list all the animals is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'there are animals',
                uponReceiving: 'a request to get all animals',
                withRequest: {
                    method: 'GET',
                    path: '/animals'
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike(
                        {
                            breed: Matchers.like("Bengali"),
                            gender: Matchers.term(
                                {generate:"Female", matcher:"Female|Male"}
                            ),
                            isVaccinated: Matchers.boolean(true),
                            name: Matchers.string("Manchas"),
                            vaccines: Matchers.eachLike("rabia", {min: 1})
                        },
                        {min: 1}
                    )
                }
            });
        });

        it("Then it should return the right data", async() => {
            const response = await AnimalController.list();
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
});