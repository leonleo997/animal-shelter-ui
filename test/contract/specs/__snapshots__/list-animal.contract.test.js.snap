// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Animal Service When a request to list all the animals is made Then it should return the right data 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "isVaccinated": true,
    "name": "Manchas",
    "vaccines": Array [
      Array [
        "rabia",
      ],
    ],
  },
]
`;
